server/router/driver.ts

```diff
@@ -12,6 +12,50 @@ import { UserType } from '../models/subscription';
 const router = express.Router();
 const tableName = 'Drivers';
 
+// Get all drivers
+router.get('/', validateUser('Admin'), (req, res) => {
+  db.getAll(res, Driver, tableName);
+});
+
+// Get all available drivers at a specific date and time
+router.get('/available', validateUser('User'), (req, res) => {
+  const { date, startTime: reqStartTime, endTime: reqEndTime } = req.query;
+  const numToDay = {
+    1: 'Mon',
+    2: 'Tue',
+    3: 'Web',
+    4: 'Thu',
+    5: 'Fri',
+  };
+  const reqDate = (numToDay as any)[moment(date as string).day()];
+
+  if ((reqStartTime as string) >= (reqEndTime as string)) {
+    res.status(400).send({ err: 'startTime must precede endTime' });
+  }
+
+  db.getAll(res, Driver, tableName, (doc: DriverType[]) => {
+    const drivers = doc.filter((driver) => {
+      const availStart = !reqDate
+        ? reqDate
+        : (driver.availability as any)[reqDate]?.startTime;
```

Comment:
we want to minimize as many `as any` casts as possible, so make sure `reqDay` is typed correctly. you can do this by adding a type to `numToDay`.

Reply:
Done, but have to import AvailabilityType.

Reply:
thats fine